### PR TITLE
OkHttp interceptors memorize the current target URL

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: retrofit2-clients/src/test/java/com/palantir/remoting3/retrofit2/TestService.java [hunk 1]
```diff
@@ -50,9 +50,6 @@
     @GET("relative")
     Call<String> getRelative();
 
-    @GET("/absolute")
```
Comment (uschi2000) [inline retrofit2-clients/src/test/java/com/palantir/remoting3/retrofit2/TestService.java:53]: Absolute URLs don't work with the redirect mechanism. I suspect we don't really need absolute URLs anyway for any of our RPC?

**Comment by uschi2000** [comment]
ping

**Comment by markelliot** [comment]
not really sure I understand what you're changing here

**Comment by uschi2000** [comment]
"not really sure I understand what you're changing here"
with this change the client memorizes the target URL across calls. e.g., after a 308 redirect, subsequent calls also go to the redirected node rather than the original one

**Pull request merged**